### PR TITLE
Prevent invalid character ranges

### PR DESCRIPTION
**Pull request by gukoff**
- Prevent invalid character ranges
- More informative error message for unsolvable regexes
- Removed tail recursion
- Fixed var name typo.

--

Throwing exception from Xeger ctor is a somewhat controversial choice.

The original library seems to not do so, and we accept the other invalid constructions such as `\w{7,1}`.
At the same time, in the function `MakeInterval` there's already a precedent of throwing.

Alternatively, I thought of of adding examples of unsolvable regexes to the README, and in the error message link to this section of README. The only examples I can think of are `[\w-.]` and `\w{7,1}`.

--

Should solve https://github.com/moodmosaic/Fare/issues/44

### CODE REVIEW

**Comment by gukoff** [comment]
@moodmosaic ping!

**Review by moodmosaic — COMMENTED** [review]
Thank you _so much_ for this! Please accept my apologies for the delay. I completely missed this.

File: Src/Fare/Xeger.cs [hunk 1]
```diff
@@ -95,29 +95,32 @@ private static int GetRandomInt(int min, int max, Random random)
 
         private void Generate(StringBuilder builder, State state)
         {
-            var transitions = state.GetSortedTransitions(true);
-            if (transitions.Count == 0)
+            while (true)
```
[the remark this answers is not in the document]
Comment (gukoff) [reply Src/Fare/Xeger.cs:98]: Why? :)

This optimization likely happens anyway, I'm just making sure that it does: https://github.com/dotnet/csharplang/issues/2304#issuecomment-908877741

**Comment by moodmosaic** [comment]
@gukoff, thank you 👍

**Pull request merged**